src/textual/driver.py

```diff
@@ -196,3 +185,73 @@ def no_automatic_restart(self) -> Iterator[None]:
 
     def close(self) -> None:
         """Perform any final cleanup."""
+
+    def open_url(self, url: str, new_tab: bool = True) -> None:
+        """Open a URL in the default web browser.
+
+        Args:
+            url: The URL to open.
+            new_tab: Whether to open the URL in a new tab.
+                This is only relevant when running via the WebDriver,
+                and is ignored when called while running through the terminal.
+        """
+        import webbrowser
+
+        webbrowser.open(url)
+
+    def deliver_binary(
+        self,
+        binary: BinaryIO | TextIO,
+        *,
+        delivery_key: str,
+        save_path: Path,
+        open_method: Literal["browser", "download"] = "download",
+        encoding: str | None = None,
+        mime_type: str | None = None,
+    ) -> None:
+        """Save the file `path_or_file` to `save_path`.
+
+        If running via web through Textual Web or Textual Serve,
+        this will initiate a download in the web browser.
+
+        Args:
+            binary: The binary file to save.
+            delivery_key: The unique key that was used to deliver the file.
+            save_path: The location to save the file to. If None,
+                the default "downloads" directory will be used. When
+                running via web, only the file name will be used.
+            open_method: *web only* Whether to open the file in the browser or
+                to prompt the user to download it. When running via a standard
+                (non-web) terminal, this is ignored.
+            encoding: *web only* The text encoding to use when saving the file.
+                This will be passed to Python's `open()` built-in function.
+                When running via web, this will be used to set the charset
+                in the `Content-Type` header.
+            mime_type: *web only* The MIME type of the file. This will be used to
+                set the `Content-Type` header in the HTTP response.
+        """
+        if isinstance(binary, BinaryIO):
+            mode = "wb"
+        else:
+            mode = "w"
+
+        def save_file_thread():
+            with open(save_path, mode) as destination_file:
+                read = binary.read
+                write = destination_file.write
+                while True:
+                    data = read(1024)
```

Comment:
This can probably be much larger. 64K would be a good starting point.

Could we use `shutil.copyfileobj` here? I imagine that is well tuned.

Reply:
I was getting typing errors with that method, so I just copied the implementation of it. I'll bump it to 64